file_server.go

```diff
@@ -29,8 +30,22 @@ func startFileServer() {
 	os.MkdirAll(opt.metaDir, os.ModePerm)
 	http.HandleFunc("/", fileServer)
 
-	log.Printf("Serving on HTTP port 4000\n")
-	log.Fatal(http.ListenAndServe(":4000", nil))
+	port := strconv.Itoa(opt.port)
+	if opt.port == 443 {
+		// In case HTTPS in port 443
+		log.Printf("Serving on HTTPS port %s\n", port)
+		err := http.ListenAndServeTLS(":"+port, "localhost.crt", "localhost.key", nil)
```

Comment:
Please check explicitly if the certificate files exists and fail with a clear message if it doesn't exist indicating what to do.